File: leetcode75/392. Is Subsequence/Solution.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {

        if (string.IsNullOrEmpty(s)) return true;
        
        int sIndex = 0;

        for (int tIndex = 0; tIndex < t.Length && sIndex < s.Length; ++tIndex)
        {
            if (s[sIndex] == t[tIndex])
            {
                sIndex++;
            } 
        }
        return sIndex == s.Length;
    }
}

File: leetcode75/392. Is Subsequence/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:

        s_index, t_index = 0, 0

        while s_index < len(s) and t_index < len(t):
            if s[s_index] == t[t_index]:
                s_index += 1
            t_index += 1
        
        return s_index == len(s)
